packages/counterfactual_nft/contracts/external/IPFS.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+pragma experimental ABIEncoderV2;
+
+
+/// @title IPFS
+/// @author Brecht Devos - <brecht@loopring.org>
+library IPFS
+{
+    bytes constant ALPHABET = '123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz';
```

Comment:
I found a few missing characters in the alphabet (uppercase `I`, uppercase `O`, lowercase `l`). Is that on purpose?
```suggestion
    bytes constant ALPHABET = '123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
```

Reply:
Makes sense in reducing common typographical errors, if human transcription were an expected method of input

Reply:
That is indeed on of the the intended goals of the [base58 alphabet](https://en.bitcoin.it/wiki/Base58Check_encoding) used by IPFS CIDs.